File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

void build_pyramid(int height);

int main(void)
{
    int height = -1;
    do
    {
        printf("Height: ");
        height = get_int();
    }
    while (height < 0 || height > 23);
    build_pyramid(height);
}

void build_pyramid(int height)
{
    for (int i = 0; i < height; i++)
    {
        for (int j = 0; j < height - 1 - i; j++)
        {
            printf(" ");
        }

        for (int k = 0; k < i + 2; k++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: pset6/mario.py
import cs50

def main():
    # Prompt user for height of pyramid until it's valid.
    while True:
        print("Height: ", end="")
        height = cs50.get_int()
        if height <= 23 and height >= 0:
            break
    print_pyramid(height)

def print_pyramid(height):
    for i in range(height):
        # Left side of pyramid.
        print(" " * (height-i-1), end="")
        print("#" * (i+2), end="")
        # Gap between pyramid.
        print("  ", end="")
        # Right side of pyramid.
        print("#" * (i+2), end="")
        # Start new row.
        print()
    
if __name__ == "__main__":
    main()